File: apps/backend/src/docuweave/domain/entities/selector.py
from dataclasses import dataclass, field


@dataclass(frozen=True)
class SelectorConfig:
    type: str  # "table_selector"
    table: str
    display_fields: tuple[str, ...] = field(default_factory=tuple)
    multi_select: bool = True


@dataclass(frozen=True)
class SelectorResult:
    selected_ids: tuple[str, ...]

File: packages/shared-types/src/selector.ts
export interface SelectorConfig {
  type: "table_selector";
  table: string;
  display_fields: string[];
  multi_select: boolean;
}

export interface SelectorOptionsRequest {
  data_source_id: string;
  selector_config: SelectorConfig;
}

export interface SelectorEvaluateRequest {
  data_source_id: string;
  selector_config: SelectorConfig;
  selected_ids: string[];
}
